test/ClassBuilder/SerializeTest.js

```diff
@@ -0,0 +1,152 @@
+/*
+ Javascript Object                               ______  ________________
+ Serialize Implementation                    __ / / __ \/  _/  _/\_____  \
+                                            / // / /_/ // /_/ /    _(__  <
+ Copyright 2014, Harold Iedema.             \___/\____/___/___/   /       \
+ --------------------------------------------------------------- /______  / ---
+ Permission is hereby granted, free of charge, to any person obtaining  \/
+ a copy of this software and associated documentation files (the
+ "Software"), to deal in the Software without restriction, including
+ without limitation the rights to use, copy, modify, merge, publish,
+ distribute, sublicense, and/or sell copies of the Software, and to
+ permit persons to whom the Software is furnished to do so, subject to
+ the following conditions:
+
+ The above copyright notice and this permission notice shall be
+ included in all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ ------------------------------------------------------------------------------
+*/
+
+/**
+ * Tests getters, setters and inheritance of instantiated class defintions.
+ */
+test('ClassBuilder:SerializeTest', function(assert) {
+
+    // BaseClass declaration with 3 public properties. We'll be extending on
+    // this class to see if the integrity of these values stay intact within
+    // the correct context.
+    var BaseClass = JOII.ClassBuilder('BaseClass', {}, {
+        'public a': 1,
+        'public noserialize b': 'foo',
+        'protected serialize c': 'bar',
```

Comment:
`'protected serialize string foo': "bar"`
vs
`'protected serializable string foo': "bar"`

The latter sounds more 'correct' to me, since it's a modifier, not a command.